ios/engine/KMEI/KeymanEngine/Classes/Migrations.swift

```diff
@@ -490,4 +501,180 @@ public enum Migrations {
       }
     }
   }
+
+  static func migrateCloudResourcesToKMPFormat() throws {
+    let userDefaults = Storage.active.userDefaults
+
+    if var userKeyboards = userDefaults.userKeyboards {
+      userKeyboards = migrateToKMPFormat(userKeyboards)
+      userDefaults.userKeyboards = userKeyboards
+    }
+
+    if var userLexicalModels = userDefaults.userLexicalModels {
+      userLexicalModels = migrateToKMPFormat(userLexicalModels)
+      userDefaults.userLexicalModels = userLexicalModels
+    }
+  }
+
+  static func migrateToKMPFormat<Resource, Package: TypedKeymanPackage<Resource>>(
+        _ resources: [Resource]) -> [Resource]
+        where Resource: KMPInitializableLanguageResource,
+              Resource == Resource.Metadata.LanguageResourceType {
+    // Step 1 - drop version numbers from all filenames.  KMP installations don't include them, and
+    // the version numbered filenames will actually interfere with migration.
+    resources.forEach { resource in
```

Comment:
On the iOS app (prior to 14.0), do previous  keyboard versions remain in the same keyboard folder?
For example, could the keyboard folder contain:
```
sil_euro_latin-1.8.js
sil_euro_latin-1.9.js
sil_euo_latin-1.9.1.js
```
and now they'd all be copied to `sil_euro_latin.js`?

Reply:
Only any versions currently referenced would be copied to `sil_euro_latin.js`.  The `primary key` is the active set of `LanguageResource`s installed, not the files themselves.